EIPS/eip-3540.md

```diff
@@ -0,0 +1,218 @@
+
+A non-exhaustive list of future improvements which could benefit from this format:
+- Including a `JUMPDEST`-table (to avoid analysis at execution time) or removing `JUMPDEST`s entirely.
+- Introducing static jumps (with relative addresses) and jump tables, and disallowing dynamic jumps at the same time.
+- Requiring code section(s) to be terminated by `STOP`. (Assumptions like this can provide significant speed improvements in interpreters, such as a speed up of ~7% seen in [evmone](https://github.com/ethereum/evmone/pull/295).)
+- Multi-byte opcodes without any workarounds.
+- Representing functions as individual code sections instead of subroutines.
+- Introducing a specific section for the [EIP-2938 Account Abstraction](./eip-2938.md) "top-level AA execution frame", simplifying the proposal.
+
+## Specification
+
+*We use [RFC2119](https://tools.ietf.org/html/rfc2119) keywords in this section.*
+
+In order to guarantee that every EOF-formatted contract in the state is valid, we need to prevent already deployed (and not validated) contracts from being recognized as such format. This will be achieved by choosing a byte sequence for the *magic* that doesn't exist in any of the already deployed contracts. To prevent the growth of the search space and to limit the analysis to the contracts existing before these changes, we split changes into two hard forks, and disallow the starting byte of the format (the first byte of the magic) in the first hard fork.
+
+### First hard fork
+
+After `block.number == HF1_BLOCK` new contract creation (via create transaction, `CREATE` or `CREATE2` instructions) results in an exceptional abort if the _code_'s first byte is `0xEF`. 
+
+#### Remarks
+
+*For purely reference purposes we call the `0xEF` byte the `FORMAT`.*
+
+The *initcode* is the code executed in the context of the *create* transaction, `CREATE`, or `CREATE2` instructions. The *initcode* returns *code* (via the `RETURN` instruction), which is inserted into the account. See section 7 ("Contract Creation") in the Yellow Paper for more information.
+
+The opcode `0xEF` is currently an unimplemented instruction, therefore: *It pops no stack items and pushes no stack items, and it causes an exceptional abort when executed.* This means *initcode* or already deployed *code* starting with this instruction will continue to abort execution.
+
+### Between the two hard forks
+
+Next step to be conducted manually: Once the first hard fork went live, all existing contracts at `block.number == HF1_BLOCK` having their first byte as the `FORMAT` are examined off-chain. The goal is to find the shortest sequence after the `FORMAT` not matching any existing contract. We expect a 1- or 2-byte sequence would suffice. We call this byte sequence the *magic*, which will be used in the second hard fork.
+
+### Second hard fork
+
```

Comment:
EIPs do not concern themselves with the hardfork coordination process.  This EIP should just have a dependency on 3541 and otherwise not bring up the "first hardfork" at all.  You may want to put a note in the backward compatibility section mentioning that there is value in having a delay between the introduction of 3541 and this EIP, but extended discussion about the migration process and timing should be left out of the EIP and put in the discussions-to link or some other external source.